dex/src/main/scala/com/wavesplatform/dex/api/MatcherWebSocketRoute.scala

```diff
@@ -1,20 +1,57 @@
 package com.wavesplatform.dex.api
 
+import java.nio.charset.StandardCharsets
 import java.time.LocalDateTime
 
+import akka.actor.ActorRef
+import akka.http.scaladsl.marshalling.ToResponseMarshaller
 import akka.http.scaladsl.model.ws.{BinaryMessage, Message, TextMessage}
-import akka.http.scaladsl.server.Route
-import akka.stream.Materializer
+import akka.http.scaladsl.server.{Directive0, Route}
 import akka.stream.scaladsl.{Flow, Sink, Source}
+import akka.stream.{CompletionStrategy, Materializer, OverflowStrategy}
+import com.google.common.primitives.Longs
+import com.wavesplatform.dex.api.PathMatchers.PublicKeyPM
 import com.wavesplatform.dex.api.http.ApiRoute
+import com.wavesplatform.dex.api.websockets.WsAddressState
+import com.wavesplatform.dex.domain.account.PublicKey
+import com.wavesplatform.dex.domain.bytes.codec.Base58
+import com.wavesplatform.dex.domain.crypto
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.{AddressActor, AddressDirectory}
 import io.swagger.annotations.Api
 import javax.ws.rs.Path
 
 import scala.concurrent.duration._
+import scala.util.Success
 
 @Path("/ws")
 @Api(value = "/web sockets/")
-case class MatcherWebSocketRoute()(implicit mat: Materializer) extends ApiRoute {
+case class MatcherWebSocketRoute(addressDirectory: ActorRef)(implicit mat: Materializer) extends ApiRoute with ScorexLogging {
+
+  private implicit val trm: ToResponseMarshaller[MatcherResponse] = MatcherResponse.toResponseMarshaller
+
+  private def accountUpdatesSource(publicKey: PublicKey): Source[TextMessage.Strict, Unit] = {
+
+    val completionMatcher: PartialFunction[Any, CompletionStrategy] = {
+      case akka.actor.Status.Success(s: CompletionStrategy) => s
+      case akka.actor.Status.Success(_)                     => CompletionStrategy.draining
+      case akka.actor.Status.Success                        => CompletionStrategy.draining
```

Comment:
Interesting... Can't find the `akka.actor.Status.Success` object.

Reply:
Ctrl + C, Ctrl + V from the akka source codes